packages/react-i18n/src/babel-plugin/test/babel-plugin.test.ts

```diff
@@ -380,6 +380,132 @@ describe('babel-plugin-react-i18n', () => {
+          }
+        `,
+        ),
+      );
+    });
+
+    it('injects proper translation function for fallback plus one case', async () => {
+      expect(
+        await transformUsingI18nBabelPlugin(
+          useI18nFixture,
+          staticImportOptionsForFile('MyComponent.tsx', 'fallbackPlusOne'),
+          {mode: 'with-explicit-paths'},
+        ),
+      ).toBe(
+        await normalize(
+          `import _en from './${TRANSLATION_DIRECTORY_NAME}/en.json';
+          import React from 'react';
+          import {useI18n} from '@shopify/react-i18n';
+
+          export default function MyComponent() {
+            const [i18n] = useI18n({
+              id: 'MyComponent_${defaultHash}',
+              fallback: _en,
+              translations(locale) {
+                if (["fr"].indexOf(locale) < 0) {
+                  return;
+                }
+
+                return (async () => {
+                  let dictionary;
+
+                  dictionary = await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n", webpackMode: "lazy-once" */ "./${TRANSLATION_DIRECTORY_NAME}/fr.json");
+
+                  return dictionary && dictionary.default;
+                })();
+              }
+            });
+            return i18n.translate('key');
+          }
+        `,
+        ),
+      );
+    });
+
+    it('injects translation function with multiple static imports', async () => {
+      expect(
+        await transformUsingI18nBabelPlugin(
+          useI18nFixture,
+          staticImportOptionsForFile('MyComponent.tsx', 'multipleTranslations'),
+          {mode: 'with-explicit-paths'},
+        ),
+      ).toBe(
+        await normalize(
+          `import _en from './${TRANSLATION_DIRECTORY_NAME}/en.json';
+          import React from 'react';
+          import {useI18n} from '@shopify/react-i18n';
+
+          export default function MyComponent() {
+            const [i18n] = useI18n({
+              id: 'MyComponent_${defaultHash}',
+              fallback: _en,
+              translations(locale) {
+                if (["de", "fr", "zh-TW"].indexOf(locale) < 0) {
+                  return;
+                }
+
+                return (async () => {
+                  let dictionary;
+
+                  switch (locale) {
+                    case "de":
+                      dictionary = await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n", webpackMode: "lazy-once" */ "./${TRANSLATION_DIRECTORY_NAME}/de.json");
+                      break;
+                    case "fr":
+                      dictionary = await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n", webpackMode: "lazy-once" */ "./${TRANSLATION_DIRECTORY_NAME}/fr.json");
+                      break;
+                    default:
+                      dictionary = await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n", webpackMode: "lazy-once" */ "./${TRANSLATION_DIRECTORY_NAME}/zh-TW.json");
```

Comment:
slightly outside the bounds of this pr but a nice change might be to avoid the intermediate variable, and thus possibly slightly simplify the output could be to chain the promise so you can immediatly return

```
const returnDefault = (mod) => mod && mod.default;

// then in each case
case: "fr":
  return await import(/*magic comment*/"./${TRANSLATION_DIRECTORY_NAME}/zh-TW.json").then(returnDefault);

```